restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java

```diff
@@ -1791,14 +1791,27 @@ private void notificationHandleForegroundUpdate(RCConnection connection)
 
    public void registerForPushNotifications(boolean itsUpdate){
       try {
-
-         if ((Boolean) parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ENABLE_PUSH_FOR_ACCOUNT)) {
-            if (storageManagerPreferences != null && RCUtils.shouldRegisterForPush(parameters, storageManagerPreferences)) {
-               new FcmConfigurationHandler(storageManagerPreferences, this).registerForPush(itsUpdate);
+          //we can have different state
```

Comment:
Can we move all this business logic inside FcmConfigurationHandler? Seems a bit of a burdain for RCDevice

Reply:
Sure.
Done.